salt/states/user.py

```diff
@@ -88,6 +88,10 @@ def _changes(name,
     attributes supported as integers only.
     '''
 
+    # Set gid to None in cases an empty string is passed
+    if gid == '':
+        gid = None
```

Comment:
Move this to `def present` right below the line 462 where `__salt__['file.group_to_gid']...` is called, since you already checking it below on lines 463-467.